src/resolve/resolved-dns-cache.c

```diff
@@ -1091,7 +1126,7 @@ int dns_cache_lookup(
                                         FLAGS_SET(j->query_flags, SD_RESOLVED_AUTHENTICATED) ? DNS_ANSWER_AUTHENTICATED : 0,
                                         NULL,
                                         query_flags,
-                                        j->until,
+                                        use_stale_data ? (current + MIN(j->rr->ttl * USEC_PER_SEC, CACHE_STALE_CLAMP_TTL_MAX_USEC)) : j->until_valid,
```

Comment:
given you have the same expression basically twice here and above. can you wrap this in a tiny helper function?

Reply:
also, the ttl is a relative value, taken to the time the record was acquired, but you add it here to the current time, that isn't right!